core/orchestrator.go

```diff
@@ -141,17 +141,28 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
 			}
 		}
 
+		// Add ticket EV to credit
+		orch.node.Accounts.Credit(manifestID, new(big.Rat).Mul(new(big.Rat).SetFrac(ticket.FaceValue, big.NewInt(1)), ticket.WinProbRat()))
+
 		if won {
 			glog.V(common.DEBUG).Infof("Received winning ticket manifestID=%v recipientRandHash=%x senderNonce=%v", manifestID, ticket.RecipientRandHash, ticket.SenderNonce)
 
 			go func(ticket *pm.Ticket, sig []byte, seed *big.Int) {
 				if err := orch.node.Recipient.RedeemWinningTicket(ticket, sig, seed); err != nil {
 					glog.Errorf("error redeeming ticket manifestID=%v recipientRandHash=%x senderNonce=%v: %v", manifestID, ticket.RecipientRandHash, ticket.SenderNonce, err)
 				}
+				// Update credit balance
 			}(ticket, tsp.Sig, seed)
 		}
 	}
 
+	// Check if there is enough credit for the manifestID to proceed
+	currBalance := orch.node.Accounts.Balance(manifestID)
+	reqBalance := new(big.Rat).SetFrac(orch.node.Recipient.MinEV(), big.NewInt(1))
+	if currBalance.Cmp(reqBalance) < 0 {
```

Comment:
Add test that this error is returned properly?

Reply:
Added `TestProcesspayment_GivenInsufficientCredit_ReturnsErr` in https://github.com/livepeer/go-livepeer/pull/956/commits/789195926a3d4628b3bc3e6acec20f4b78035e68